src/SFA.DAS.FAT.Application.UnitTests/Services/WhenGettingCourseProviderDetails.cs

```diff
@@ -0,0 +1,105 @@
+﻿using AutoFixture.NUnit3;
+using Microsoft.Extensions.Options;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.FAT.Application.Services;
+using SFA.DAS.FAT.Domain.Configuration;
+using SFA.DAS.FAT.Domain.Courses;
+using SFA.DAS.FAT.Domain.Courses.Api.Requests;
+using SFA.DAS.FAT.Domain.Interfaces;
+using SFA.DAS.Testing.AutoFixture;
+
+namespace SFA.DAS.FAT.Application.UnitTests.Services;
+
+public class WhenGettingCourseProviderDetails
+{
+    [Test]
+    [MoqAutoData]
+    public async Task Then_The_Correct_Response_Is_Returned_From_The_Api(
+        string baseUrl,
+        int ukprn,
+        int larsCode,
+        string location,
+        int? distance,
+        Guid shortlistUserId,
+        CourseProviderDetailsModel CourseProviderDetailsResponse,
+        [Frozen] Mock<IOptions<FindApprenticeshipTrainingApi>> findApprenticeshipTrainingApiConfigurationMock,
+        [Frozen] Mock<IApiClient> mockApiClient,
+        CourseService sut
+    )
+    {
+        mockApiClient.Setup(x => x.Get<CourseProviderDetailsModel>(
+            It.Is<GetCourseProviderDetailsApiRequest>(request =>
+                request.GetUrl.Contains(ukprn.ToString()) &&
+                request.GetUrl.Contains(larsCode.ToString()) &&
+                request.GetUrl.Contains(location) &&
+                request.GetUrl.Contains(location) &&
+                request.GetUrl.Contains(shortlistUserId.ToString())
+            )
+        )).ReturnsAsync(CourseProviderDetailsResponse);
+
+        var result = await sut.GetCourseProvider(
+            ukprn, 
+            larsCode, 
+            location, 
+            distance, 
+            shortlistUserId
+        );
+
+        Assert.Multiple(() =>
+        {
+            Assert.That(result.Ukprn, Is.EqualTo(CourseProviderDetailsResponse.Ukprn));
```

Comment:
I mcould be wrong by I suspect result.Should().BeEquiivalentTo(CourseProviderDetailsResponse) will do the same and be cleaner